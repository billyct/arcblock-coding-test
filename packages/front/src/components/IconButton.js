import PropTypes from 'prop-types'

const IconButton = ({children, icon, iconPlacement, ...props}) => (
  <button
    className='flex items-center disabled:opacity-50'
    {...props}
  >
    {iconPlacement === 'left' && icon}
    <span className='mx-2 font-bold'>{children}</span>
    {iconPlacement === 'right' && icon}
  </button>
)

IconButton.prototype = {
  icon: PropTypes.any.isRequired,
  iconPlacement: PropTypes.oneOf(['left', 'right']),
}

IconButton.defaultProps = {
  iconPlacement: 'right',
}

export default IconButton
