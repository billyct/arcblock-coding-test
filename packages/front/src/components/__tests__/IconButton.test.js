import {render} from '@testing-library/react'

import IconButton from '../IconButton'

describe.each([
  [
    {
      icon: 'icon',
      iconPlacement: 'left',
      children: 'button'
    },
    'iconbutton',
  ],

  [
    {
      icon: 'icon',
      children: 'button'
    },
    'buttonicon',
  ],
])('test IconButton Component with %o', (attribute, textContent) => {
  it(`should haveTextContent(${textContent})`, function () {
    const {container} = render(<IconButton {...attribute}/>)
    expect(container).toHaveTextContent(textContent)
  })

  it('should render correct', function () {
    const {container} = render(<IconButton {...attribute}/>)
    expect(container).toMatchSnapshot()
  })
})
