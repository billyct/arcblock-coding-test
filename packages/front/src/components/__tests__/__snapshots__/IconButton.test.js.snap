// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test IconButton Component with { icon: 'icon', children: 'button' } should render correct 1`] = `
<div>
  <button
    class="flex items-center disabled:opacity-50"
  >
    <span
      class="mx-2 font-bold"
    >
      button
    </span>
    icon
  </button>
</div>
`;

exports[`test IconButton Component with { icon: 'icon', iconPlacement: 'left', children: 'button' } should render correct 1`] = `
<div>
  <button
    class="flex items-center disabled:opacity-50"
  >
    icon
    <span
      class="mx-2 font-bold"
    >
      button
    </span>
  </button>
</div>
`;
